src/bin/enclave-runc-wrapper.rs

```diff
@@ -0,0 +1,241 @@
+use k8s_openapi::api::core::v1;
+use std::convert::TryFrom;
+
+// TODO: To be used for cri-o.
+// static ANNOTATION_K8S_LABELS: &str = "io.kubernetes.cri-o.Labels";
+
+// static LABEL_NAMESPACE: &str = "io.kubernetes.pod.namespace";
+static LABEL_POLICY_ENFORCE: &str = "pod-security.kubernetes.io/enforce";
+// static LABEL_POLICY_AUDIT: &str = "pod-security.kubernetes.io/audit";
+// static LABEL_POLICY_WARN: &str = "pod-security.kubernetes.io/warn";
+
+static ANNOTATION_CONTAINERD_LOG_DIRECTORY: &str = "io.kubernetes.cri.sandbox-log-directory";
+static ANNOTATION_CONTAINERD_SANDBOX_ID: &str = "io.kubernetes.cri.sandbox-id";
+
+#[derive(thiserror::Error, Debug)]
+enum NewContainerError {
+    #[error("could not retrieve the runc status")]
+    Status(#[from] std::io::Error),
+
+    #[error("could not format")]
+    Format(#[from] std::fmt::Error),
+
+    #[error("could not convert bytes to utf-8 string")]
+    Utf8(#[from] std::string::FromUtf8Error),
+
+    #[error("could not parse JSON")]
+    Json(#[from] serde_json::Error),
```

Comment:
So these `thiserror` declarations extend `std::error::Error` by default (if I understand correctly), and we essentially want these variants to extend a _different_ error type?

Reply:
My usecase for `thiserror` is accepting more than one type of error which can be thrown by the function. And I'm occasionally using it to define my own errors.